xmtp_mls/src/groups/decoded_message.rs

```diff
@@ -0,0 +1,94 @@
+use xmtp_content_types::{
+    attachment::Attachment, read_receipt::ReadReceipt, remote_attachment::RemoteAttachment,
+    transaction_reference::TransactionReference,
+};
+use xmtp_db::group_message::{DeliveryStatus, GroupMessageKind};
+use xmtp_proto::xmtp::mls::message_contents::{
+    ContentTypeId, EncodedContent, GroupMembershipChanges, GroupUpdated,
+    content_types::{MultiRemoteAttachment, ReactionV2},
+};
+
+#[derive(Debug, Clone)]
+pub struct Reply {
+    // The original message that this reply is in reply to.
+    // This goes at most one level deep from the original message, and won't happen recursively if there are replies to replies to replies
+    pub in_reply_to: Option<Box<DecodedMessage>>,
+    pub content: Box<MessageBody>,
+}
+
+#[derive(Debug, Clone)]
+pub enum ReactionAction {
+    Added,
+    Removed,
+}
+
+#[derive(Debug, Clone)]
+pub enum ReactionSchema {
+    Unicode,
+    Shortcode,
+    Custom,
+}
+
+#[derive(Debug, Clone)]
+pub struct Reaction {
```

Comment:
I ended up killing this struct in an upstack PR, after I figured out how to deal with recursive content.